src/Linkify.jsx

```diff
@@ -12,18 +12,54 @@ class Linkify extends React.Component {
     component: React.PropTypes.any,
     properties: React.PropTypes.object,
     urlRegex: React.PropTypes.object,
-    emailRegex: React.PropTypes.object
+    emailRegex: React.PropTypes.object,
+    handlers: React.PropTypes.arrayOf(
+      React.PropTypes.shape({
+        prefix: React.PropTypes.string.isRequired,
+        validate: React.PropTypes.func.isRequired,
+        normalize: React.PropTypes.func.isRequired
+      })
+    )
   }
 
   static defaultProps = {
     component: 'a',
     properties: {},
+    handlers: []
+  }
+
+  componentDidMount() {
+    this.addCustomHandlers();
+  }
+
+  componentDidUpdate(nextProps) {
+    if (this.props.handlers !== nextProps.handlers) {
+      this.addCustomHandlers();
+    }
+  }
+
+  addCustomHandlers() {
+    const { handlers } = this.props;
+
+    if (handlers.length) {
+      this.linkify = new LinkifyIt();
+      this.linkify.tlds(tlds);
+
+      handlers.forEach(handler => {
+        this.linkify.add(handler.prefix, {
+          validate: handler.validate,
+          normalize: handler.normalize
+        });
+      });
+    }
   }
 
   parseCounter = 0
 
   getMatches(string) {
-    return linkify.match(string);
+    const linkifyInstance = this.linkify || linkify;
```

Comment:
What about for use-cases of supporting both individually and globally defined handlers? We might want some sort of merging of handlers here.